File: src/main/scala/zpj/bean/Person.scala
package zpj.bean

/**
  * Created by PerkinsZhu on 2017/9/23 10:39. 
  */
/*case class Person(name:String,age:Int)
case class Student(override val name:String, override val age:Int, classs:String) extends Person(name,age)*/

case class Person(name: String, age: Int)
class Student private(name: String, age: Int) extends Person(name, age) {
  def show(): String = {
    "Hello"
  }

  def run(): Unit = {

  }

  def runWithMe(me: String): Unit = {

  }

}
class Teacher(name: String, age: Int, isMale: Boolean, salary: Float, student: Student) extends Person(name, age) {

  override def toString = s"Teacher($name,$age,$isMale,$salary,$student)"
}

File: src/main/java/basetest/bean/Person.java
package basetest.bean;

/**
 * Created by PerkinsZhu on 2019/5/11 11:47
 **/
public class Person {
    protected final String name = "jack";
//    protected static Intger age;

    class Inner{}

    static  class StaticInner{}
    public void testInner(){
        Person p = new Person();
        Inner inner = new Inner();
        Person.Inner in2 = p.new Inner();
        System.out.println(inner);
        System.out.println(in2);
        Person.StaticInner staticInner = new Person.StaticInner();
        Person.StaticInner staticInner2 = new Person.StaticInner();
        System.out.println(staticInner);
        System.out.println(staticInner2);


    }

    final String work (){

        return "";
    }
}
